src/rendering/MaterialParser/MaterialParser.cc

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <cstddef>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "MaterialParser.hh"
+
+using namespace gz;
+using namespace sim;
+
+/////////////////////////////////////////////////
+MaterialParser::MaterialParser()
+  : configLoader()
+{
+}
+
+/////////////////////////////////////////////////
+void MaterialParser::Load()
+{
+  ConfigLoader::loadMaterialFiles(&this->configLoader);
+}
+
+/////////////////////////////////////////////////
+std::optional<MaterialParser::MaterialValues> MaterialParser::GetMaterialValues(
+  const std::string& material)
+{
+  MaterialValues values;
```

Comment:
This should be a `std::optional` and should only have a value if the material is found in the search below.